src/NATS.Client.Core/NatsConnection.RequestSub.cs

```diff
@@ -1,31 +1,167 @@
 using System.Buffers;
+using System.Runtime.CompilerServices;
 using NATS.Client.Core.Internal;
 
 namespace NATS.Client.Core;
 
 public partial class NatsConnection
 {
-    internal async ValueTask<NatsSub> RequestSubAsync(
+    public async ValueTask<NatsMsg<TReply?>?> RequestAsync<TRequest, TReply>(
```

Comment:
Are we definitely set on making moving these to `NatsConnection`?  Or should we try to convert them to extension methods that extend `INatsConnection` instead?

The extension methods could be nice for someone else who implements `INatsConnection` but I'm not sure if that will be common

Reply:
I've made changes to keep the four extension methods that don't need to be instance members as extension methods and only moved those four to the interface that access internal methods (i.e. `RequestSubAsync`).

I think outside of tests it's highly unlikely that anyone want's to implement INatsConnection.